.github/workflows/build-and-test.yml

```diff
@@ -1,11 +1,39 @@
-name: test
+name: Build and test
 on:
+  schedule:
+    - cron: '18 22 * * 6'
   push:
     branches:
       - main
   pull_request:
 
 jobs:
+  build:
+    runs-on: ubuntu-22.04
+    steps:
+      # Set fetch-depth: 0 to fetch commit history and tags for use in version calculation
+      - name: Check out code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        if: github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name == github.repository
+        with:
+          username: ${{ secrets.DOCKERHUB_READ_USER }}
+          password: ${{ secrets.DOCKERHUB_READ_TOKEN }}
+
+      - name: Build with Gradle
+        uses: hypertrace/github-actions/gradle@main
+        with:
+          args: assemble dockerBuildImages
+
+      - name: Run Trivy vulnerability scanner
```

Comment:
should we add `if` condition here as well?

Reply:
hmm let me go check inside the action - from the outside it appears safe to run on a fork

Reply:
So the upload would presumably fail - perhaps change output mode for forks to log?

Reply:
I had changed the local one here to 
```
output-mode: ${{ (github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name == github.repository) && 'github' || 'log' }}
```
but if we can put that defaulting into the action it'd be cleaner.

Reply:
backed my local change out in favor of the shared action change